bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableConnection.java

```diff
@@ -304,6 +304,7 @@ private ExecutorService getBatchPool() {
       // shared HTable thread executor not yet initialized
       synchronized (this) {
         if (batchPool == null) {
+          int minThreads = getMinThreads();
```

Comment:
I used Math.min to solve this problem. I don't know that it's worth spending that much time and effort to do better, but I will if you think it's important.